addons/print/wizard/print_order_wizard.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+from openerp import api, fields, models, _
+from openerp.exceptions import Warning
+
+
+class print_order_wizard(models.TransientModel):
+
+    _name = 'print.order.wizard'
+    _rec_name = 'id'
+
+    @api.model
+    def _default_currency(self):
+        return self.env.user.company_id.currency_id
+
+    @api.model
+    def _default_print_provider(self):
+        return self.env['ir.values'].get_default('print.order', 'provider_id')
+
+    @api.one
+    @api.depends('print_order_line_wizard_ids')
+    def _compute_nbr_line(self):
+        self.nbr_lines = len(self.print_order_line_wizard_ids)
+
+    @api.one
+    @api.depends('print_order_line_wizard_ids')
+    def _compute_error_message(self):
+        message = False
+        unsendable = len(self.print_order_line_wizard_ids.filtered(lambda l: l.state != 'ok'))
+        if unsendable:
+            if len(self.print_order_line_wizard_ids) == 1:
+                current_state = self.print_order_line_wizard_ids[0].state
+                state_selection = dict(self.env['print.order.line.wizard']._default_selection_state())
+                message = _("The document you want to send contains the following mistake : %s") % (state_selection.get(current_state, "Unknow error."),)
+            else:
+                message = _("At least one of the document you want to send contains a mistake.")
+        self.error_message = message
+
+    # --------------------------------------------------
+    # MODEL FIELDS
+    # --------------------------------------------------
+    ink = fields.Selection([('BW', 'Black & White'), ('CL', 'Colour')], "Ink", default='BW')
+    paper_weight = fields.Integer("Paper Weight", default=80, readonly=True)
+    provider_id = fields.Many2one('print.provider', 'Print Provider', required=True, default=_default_print_provider)
+    provider_balance = fields.Float("Provider Credit", digits=(16, 2))
+    provider_environment = fields.Selection([('test', 'Test'), ('production', 'Production')], "Environment", default=False)
```

Comment:
Standard fields + onchange or directly related ? Can the use change those values ?